sceptre/plan/actions.py

```diff
@@ -555,6 +555,23 @@ def execute_change_set(self, change_set_name):
         status = self._wait_for_completion()
         return status
 
+    def change_set_creation_failed_due_to_no_changes(self, reason: str) -> bool:
```

Comment:
I think that consistency with the rest of the code base here would mean we don't provide type hints—as much as I also like type hints! Normally, rather, the return type is indicated in the doc string. (See elsewhere in the file for examples.) (I daresay `no_change_substrings` below could be a list to be more consistent too, although I didn't look at it closely and have no strong opinion.)